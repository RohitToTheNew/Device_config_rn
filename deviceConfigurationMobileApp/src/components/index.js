import Header from './header';
import SliderComponent from './slider';
import ErrorWindow from './errorWindow';
import CustomLoader from './customLoader';
import CustomButton from './customButton';
import CustomTextInput from './customTextInput';
import RefreshComponent from './refreshComponent';
import ManageSchool from './floatingButtonManageSchool';
import ManageSchoolHeaderButton from './ManageSchoolHeader';

export {
  Header,
  CustomLoader,
  CustomButton,
  ErrorWindow,
  ManageSchool,
  SliderComponent,
  CustomTextInput,
  RefreshComponent,
  ManageSchoolHeaderButton
};
