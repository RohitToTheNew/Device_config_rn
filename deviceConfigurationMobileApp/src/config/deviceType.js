const deviceType = {
  ms700: 'MS-700',
  infoView: 'InfoviewPlayer',
  cza1300:'CZA1300'
};

export default deviceType;
