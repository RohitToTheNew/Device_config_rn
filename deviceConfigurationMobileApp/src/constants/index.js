import UUIDMappingMS700 from './uuidMapping-ms700';
import UUIDMappingInfoView from './uuidMapping-infoview';
import DefaultCredentials from './defaultCredentials';
const boudRate = [9600, 19200, 38400, 57600, 115200]

export {DefaultCredentials, UUIDMappingMS700, UUIDMappingInfoView, boudRate};
