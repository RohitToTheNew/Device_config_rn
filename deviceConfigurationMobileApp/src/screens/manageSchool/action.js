const identifire = 'manage-school-log: ';
import { Alert } from 'react-native';
import RNFS from 'react-native-fs';
import Mailer from 'react-native-mail';
import Utils from '../../utils';
import { translate } from '../../translations/translationHelper';

/**
 * function to get saved Schools data from local storage
 */
function getSavedData() {
  const result = RNFS.readFile(RNFS.DocumentDirectoryPath + '/test.txt', 'utf8');
  return result;
}

/**
 * function for formatting the rooms data to desired CSV format
 * @param {object} parms object containing the data of rooms
 * @returns formatted CSV data
 */
function getDataInDesiredFormate(parms) {
  let CSVData = [];
  parms.forEach(room => {
    for (const key in room) {
      if (key === 'devices') {
        room[key].forEach(roomData => {
          let obj = {};
          obj.driverName = roomData.deviceType;
          obj.deviceName = roomData.deviceName;
          obj.ip = roomData.ipAddress || '';
          obj.mac = roomData.macAddress;
          obj.roomName = room['title'];
          CSVData.push(obj);
        });
      }
    }
  });
  return CSVData;
}

/**
 * function to save schools list to local storage
 * @param {object} schools object containing schools list
 * @returns 
 */
const writeDataTolocalStorage = async schools => {
  const data = JSON.stringify(schools);
  const path = RNFS.DocumentDirectoryPath + '/test.txt';
  try {
    await RNFS.writeFile(path, data, 'utf8');
    return `FILE WRITTEN!`;
  } catch (err) {
    Utils.Log(Utils.logType.error, 'error in writeDataTolocalStorage', err)
    return `Error while writing`;
  }
};

/**
 * function to write schools data to local storage
 * @param {object} schools object containing data of schools
 * @param {string} fileName name of file containing data of schools
 * @returns error(if any) during wiriting to local storage OR success message if writing is successful
 */

const writeCSVDataTolocalStorage = async (schools, fileName) => {
  const resultCSV = convertToCSV(schools);
  const pathToWrite = `${RNFS.DocumentDirectoryPath}/${fileName}.csv`;
  try {
    await RNFS.writeFile(pathToWrite, resultCSV, 'utf8');
    return `FILE WRITTEN!`;
  } catch (err) {
    Utils.showToast(translate('errorWhileWriting.'));
    return `Error while writing`;
  }
};
/**
 * function to attach generated CSV in email and open email client
 * @param {string} fileName name of file containing data of schools
 */
function sendCSVAsMail(schoolName, fileName) {
  try {
    Mailer.mail(
      {
        subject: `${schoolName} Devices`,
        isHTML: true,
        attachments: [
          {
            type: 'csv',
            path: `${RNFS.DocumentDirectoryPath}/${fileName}.csv`, // The absolute path of the file from which to read data.
          },
        ],
      },

      (error, event) => {
        Alert.alert(
          error,
          event,
          [
            {
              text: 'Ok',
              onPress: () => { },
            },
            {
              text: 'Cancel',
              onPress: () => { },
            },
          ],
          { cancelable: true },
        );
      },
    );
  } catch (error) {
    Utils.Log(Utils.logType.error, 'error in sendCSVAsMail', error);
  }
}

/**
 * function to handle the 'Export CSV' action for school
 * @param {object} parms object containig data of school
 */
async function handleExportCSV(parms) {
  try {
    const schoolName = parms?.schoolTitle;

    let month = new Date().getMonth() + 1;
    let date = new Date().getDate();
    const year = new Date().getFullYear();

    if (date < 10) date = '0' + date;
    if (month < 10) month = '0' + month;

    const hour = new Date().getHours();
    const minute = new Date().getMinutes();
    const second = new Date().getSeconds();

    const currentDate = `${month}-${date}-${year}`;
    const currentTime = `${hour}H${minute}M${second}S`;

    const fileName = `${schoolName}_Devices_${currentDate}_${currentTime}`
      .toString()
      .trim();
    const CSVData = getDataInDesiredFormate(parms?.rooms);
    await writeCSVDataTolocalStorage(CSVData, fileName);
    sendCSVAsMail(schoolName, fileName);
  } catch (error) {
    Utils.Log(Utils.logType.error, 'handleExportCSV', error);
  }
}

/**
 * function to convert the object data to desired CSV format
 * @param {object} arr array of rooms details
 * @returns
 */
function convertToCSV(arr) {
  const array = [Object.keys(arr[0])].concat(arr);
  return array
    .map(it => {
      return Object.values(it).toString();
    })
    .join('\n');
}

/**
 * function to show alert when delete button is pressed
 * @param {string} schoolTitle title of school
 * @param {string} schoolId school id
 * @param {function} setSchoolTitle function to set school title
 * @param {function} setSchoolId function to set school id
 * @param {function} setPopupToDeleteDevice function to set popup to show
 */
function deleteSchoolAlert(
  schoolTitle,
  schoolId,
  setSchoolTitle,
  setSchoolId,
  setPopupToDeleteDevice,
) {
  setSchoolTitle(schoolTitle);
  setSchoolId(schoolId);
  setPopupToDeleteDevice(true);
}

/**
 * function to delete school
 * @param {string} schoolID school id
 * @param {object} savedData object containing saved schools data
 * @param {function} setSavedData function to set saved schools data after change
 * @param {*} setPopupToDeleteDevice function to set popup to hide
 */
async function deleteSchool(
  schoolID,
  savedData,
  setSavedData,
  setPopupToDeleteDevice,
) {
  try {
    const filteredData = savedData.filter(school => school?.id !== schoolID);
    await writeDataTolocalStorage(filteredData);
    setSavedData(filteredData);
    setPopupToDeleteDevice(false);
  } catch (error) {
    Utils.Log(Utils.logType.error, 'deleteSchool', error);
  }
}
export {
  getSavedData,
  getDataInDesiredFormate,
  writeCSVDataTolocalStorage,
  sendCSVAsMail,
  handleExportCSV,
  deleteSchoolAlert,
  deleteSchool,
  writeDataTolocalStorage,
};
