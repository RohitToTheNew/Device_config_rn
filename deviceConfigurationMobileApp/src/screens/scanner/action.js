const identifire = 'scanner-action log: ';
import { check, request, RESULTS, PERMISSIONS } from 'react-native-permissions';
import { RNCamera } from 'react-native-camera';
import utils from '../../utils';
import { translate } from '../../translations/translationHelper';
import RNFS from 'react-native-fs';

/**
 * On click Enter details manually and redirect user to Add device screen
 * @param {object} navigation it represents navigation object
 * @returns object
 */
function actionHandleManualDetails(navigation) {
  return () => {
    navigation.navigate('addDevice');
  };
}

/**
 * function to read local file data from storage
 * @returns file details read from storage
 */
const readFileFromLocalStorage = async () => {
  let result = RNFS.readFile(RNFS.DocumentDirectoryPath + '/test.txt', 'utf8');
  return result;
};

/**
 * function to check if the device details encoded into the QR code is having valid keys
 * @param {object} obj object containing the scanned data from QR code
 * @returns boolean indicating weather details are valid or not
 */
const isValidObject = (obj) =>
  obj?.serialNumber && obj?.serialNumber !== '' &&
  obj?.macAddress && obj?.macAddress !== '' &&
  obj?.deviceType && obj?.deviceType !== '';

/**
 * function to check if the device details encoded into the QR code is having valid keys
 * @param {object} obj object containing the scanned data from QR code
 * @returns boolean indicating weather details are valid or not
 */
const isValidDetails = (obj) =>
  obj?.serial && obj?.serial !== '' &&
  obj?.mac && obj?.mac !== '' &&
  obj?.device && obj?.device !== '';

/**
 * On successful scan
 * @param {object} navigation it represents navigation object
 * @param {function} setInvalidCode used to set invalid code boolean value
 * @param {object} e contains scan result
 */
function actionOnSuccess(navigation, setInvalidCode, e) {
  const typeToCheck = utils.isIOS ? 'org.iso.QRCode' : "QR_CODE"
  if (e.type === typeToCheck) {
    try {
      const scannedString = JSON.parse(e?.data);
      console.log('scanned string', scannedString)
      if (isValidObject(scannedString) || isValidDetails(scannedString)) {
        navigation.replace('addDevice', {
          mac: scannedString?.macAddress || scannedString?.mac,
          serialNumber: scannedString?.serialNumber || scannedString?.serial,
          deviceType: scannedString?.deviceType || scannedString?.device,
        });
      } else {
        setInvalidCode(true);
      }
    } catch (e) {
      setInvalidCode(true);
    }
  } else {
    setInvalidCode(true);
  }
}

/**
 * Click on close icon
 * @param {object} navigation it represents navigation object
 * @returns object
 */
function actionHandleClose(navigation) {
  return () => {
    navigation.goBack();
  };
}

/**
 * Click on flash icon
 * @param {boolean} isFlashOn used to identify if flash is on/off
 * @param {function} setFlash used to set flash on/off value
 * @param {function} setIsFlashOn used to set flash on value
 * @returns object
 */
function actionHandleFlash(isFlashOn, setFlash, setIsFlashOn) {
  return () => {
    if (isFlashOn) {
      setFlash(RNCamera.Constants.FlashMode.off);
      setIsFlashOn(false);
    } else {
      setFlash(RNCamera.Constants.FlashMode.torch);
      setIsFlashOn(true);
    }
  };
}

/**
 * To ask camera permission
 * @returns camera permission
 */
async function actionUseEffect() {

  try {
    const result = await check(PERMISSIONS.IOS.CAMERA);
    if (result !== RESULTS.GRANTED) {
      const askCameraPermission = await request(PERMISSIONS.IOS.CAMERA);
      return askCameraPermission;
    } else {
      utils.Log(utils.logType.log, identifire, 'Camera permisson granted');
    }
  } catch (Ex) {
    utils.showToast(translate('error'), translate('permissionsException'));
  }
}
export {
  actionHandleManualDetails,
  actionOnSuccess,
  actionHandleClose,
  actionUseEffect,
  actionHandleFlash,
};
