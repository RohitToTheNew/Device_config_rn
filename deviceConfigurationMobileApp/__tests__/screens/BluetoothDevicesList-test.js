import React from 'react';
import renderer from 'react-test-renderer';
import {Provider, useDispatch, useSelector} from 'react-redux';
import {store} from '../../src/store/configureStore';
import {RefreshComponent} from '../../src/components';
import blemanager from '../../src/config/bleManagerInstance';
import {updateAppModalFields} from '../../src/services/app/action';
import {
  render,
  screen,
  fireEvent,
  cleanup,
} from '@testing-library/react-native';
import {
  getDeviceType,
  handleDevicesFound,
  handleDeviceTypeBasedIdentify,
  handleIdentifyDevice,
  identifyDeviceFromList,
  navigateToDevice,
  performAuthProcess,
} from '../../src/services/bleDevices/action';
import BluetoothDevicesList from '../../src/screens/bluetoothDevicesList';
import {updateBleDevicesFields} from '../../src/services/bleDevices/action';
import {updateAuthDevices} from '../../src/services/authDevices/action';
import DeviceType from '../../src/config/deviceType';
import BleManager from '../../src/config/bleManagerInstance';
import { UUIDMappingMS700 } from '../../src/constants';
import base64 from 'react-native-base64';
import deviceType from '../../src/config/deviceType';

afterEach(cleanup);
jest.mock('../../src/components/disconnectionModal', () => {
  return {
    __esModule: true,
    default: jest.fn(() => null),
  };
});
const mockDispatch = jest.fn();
const mockBluetoothDevices = [{id: '1', localName: 'Brx-ajdhas8asn',startPairing:true}];
useDispatch.mockReturnValue(mockDispatch);
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn().mockImplementation(mockDispatch),
}));

jest.mock('@react-navigation/native', () => ({
  useIsFocused: jest.fn(),
}));

beforeEach(() => {
  useSelector.mockImplementation(selector => {
    const initialState = {
      ble: {bluetoothDevices: mockBluetoothDevices},
      app: {bluetoothState: 'PoweredOn', showDisconnectedModal: false},
      authDevices: {connectedDevice: {id: 1, localName: 'Brx-asdkasbj'}},
    };
    if (selector) {
      return selector(initialState);
    }
  });

  useDispatch.mockReturnValue(mockDispatch);
});

describe('on landing to bluetooth devices list', () => {
  let tree;

  beforeEach(() => {
    tree = renderer.create(<BluetoothDevicesList />).toJSON();
  });
  it('should render devices list correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});

describe('Bluetooth devices list found during discovery', () => {
  let component, scanSpy;
  beforeEach(() => {
    component = (
      <>
        <RefreshComponent />
        <BluetoothDevicesList />
      </>
    );
    useSelector.mockImplementation(selector => {
      const initialState = {
        ble: {bluetoothDevices: mockBluetoothDevices},
        app: {bluetoothState: 'PoweredOn', showDisconnectedModal: false},
        authDevices: {connectedDevice: {}},
      };
      if (selector) {
        return selector(initialState);
      }
    });
    getByTestId = render(component).getByTestId;
    scanSpy = jest.spyOn(blemanager, 'startDeviceScan');
    store.dispatch(updateAppModalFields('isLoading', false));
  });
  it('should render devices list screen', () => {
      expect(screen.getByTestId('devicesList')).toBeTruthy();
      expect(screen.getByTestId('bluetoothDeviceItem')).toBeTruthy();
  });
});

describe('No Bluetooth devices found during discovery', () => {
  let component, getByTestId;
  beforeEach(() => {
    component = (
        <BluetoothDevicesList />
    );
    useSelector.mockImplementation(selector => {
      const initialState = {
        ble: {bluetoothDevices: []},
        app: {bluetoothState: 'PoweredOn', showDisconnectedModal: false},
        authDevices: {connectedDevice: {}},
      };
      if (selector) {
        return selector(initialState);
      }
    });
    getByTestId = render(component).getByTestId;
  });
  it('should render no device screen', () => {
    expect(screen.getByTestId('noDevicesComponent')).toBeTruthy();
  });
});

describe('performAuthProcess function', () => {
  let navigation, navigationSpy;
  beforeEach(() => {
    navigation = {navigate: jest.fn()};
    navigationSpy = jest.spyOn(navigation, 'navigate');
    store.dispatch(
      updateBleDevicesFields('bluetoothDevices', [{localName: 'Brx-emulator'}]),
    );
  });
  it('should take device id and perform auth process', async () => {
    store.dispatch(updateAuthDevices('authenticatedDevices', []));
    await store.dispatch(performAuthProcess({deviceId:1,deviceType:DeviceType.ms700}, navigation, ()=>{}));
    expect(navigationSpy).toBeCalledWith('SetPasscode',{isUpdateScreen: false, devicePressed:{deviceId:1,deviceType:DeviceType.ms700}});
  });
});

describe('handleIdentifyDevice function for ms700', () => {
  let writeSpy;
  beforeEach(() => {
    writeSpy = jest.spyOn(BleManager, 'writeCharacteristicWithResponseForDevice')
    store.dispatch(
      updateBleDevicesFields('bluetoothDevices', [{localName: 'Brx-emulator'}]),
    );
  });
  it('should trigger identify device alarm on BLE device', async () => {
    jest.spyOn(base64,'encode').mockImplementation(()=>{
      return 'beep-tone'
    })
    await store.dispatch(handleIdentifyDevice({id:1, deviceType:DeviceType.ms700}));
    expect(writeSpy).toBeCalledWith(1,UUIDMappingMS700.rootServiceUDID, UUIDMappingMS700.identifyDevice,'beep-tone');
  });
});

describe('handleIdentifyDevice function for infoview', () => {
  let writeSpy;
  beforeEach(() => {
    writeSpy = jest.spyOn(BleManager, 'writeCharacteristicWithResponseForDevice')
    store.dispatch(
      updateBleDevicesFields('bluetoothDevices', [{localName: 'Brx-emulator'}]),
    );
  });
  it('should trigger identify device alarm on BLE device', async () => {
    jest.spyOn(base64,'encode').mockImplementation(()=>{
      return 'pop-up'
    })
    await store.dispatch(handleIdentifyDevice({id:1, deviceType:DeviceType.infoView}));
    expect(writeSpy).toBeCalledWith(1,UUIDMappingMS700.rootServiceUDID, UUIDMappingMS700.identifyDevice,'beep-tone');
  });
});

describe('handleIdentifyDevice function for CZA1300', () => {
  let writeSpy;
  beforeEach(() => {
    writeSpy = jest.spyOn(BleManager, 'writeCharacteristicWithResponseForDevice')
    store.dispatch(
      updateBleDevicesFields('bluetoothDevices', [{localName: 'Brx-emulator (cza1300)'}]),
    );
  });
  it('should trigger identify device alarm on BLE device', async () => {
    jest.spyOn(base64,'encode').mockImplementation(()=>{
      return 'blink-leds'
    })
    await store.dispatch(handleIdentifyDevice({id:1, deviceType:DeviceType.cza1300}));
    expect(writeSpy).toBeCalledWith(1,UUIDMappingMS700.rootServiceUDID, UUIDMappingMS700.identifyDevice,'blink-leds');
  });
});

describe('getDeviceType function', () => {
  let readSpy;
  beforeEach(() => {
    readSpy = jest.spyOn(BleManager, 'readCharacteristicForDevice')
    store.dispatch(
      updateBleDevicesFields('bluetoothDevices', [{localName: 'Brx-emulator', deviceType:'',id:1}]),
    );
  });
  it('should read device type from BLE device', async () => {
    jest.spyOn(BleManager,'connectToDevice').mockResolvedValueOnce({discoverAllServicesAndCharacteristics: jest.fn()})
    jest.spyOn(BleManager,'readCharacteristicForDevice').mockResolvedValueOnce({value: 'TVMtNzAw'})
    jest.spyOn(base64,'decode').mockImplementation(()=>{
      return 'MS-700'
    })
    await store.dispatch(getDeviceType({id:1,deviceType:''}));
    expect(readSpy).toBeCalledWith(1,UUIDMappingMS700.rootServiceUDID, UUIDMappingMS700.deviceType);
  });
});

describe('identifyDeviceFromList function', () => {
  let writeSpy, navigation;
  beforeEach(() => {
    writeSpy = jest.spyOn(BleManager, 'writeCharacteristicWithResponseForDevice')
    navigation = {navigate: jest.fn()};
    store.dispatch(
      updateBleDevicesFields('bluetoothDevices', [{localName: 'Brx-emulator', deviceType:'',id:1}]),
    );
    jest.spyOn(BleManager,'connectToDevice').mockResolvedValueOnce({discoverAllServicesAndCharacteristics: jest.fn(), })
  });
  it('should trigger identify device alarm on BLE device', async () => {
    await store.dispatch(identifyDeviceFromList({id: 1, localName: 'Brx-asdkasbj', deviceType:DeviceType.ms700},navigation));
    expect(writeSpy).toBeCalledWith(1,UUIDMappingMS700.rootServiceUDID, UUIDMappingMS700.identifyDevice,'beep-tone');
  });
});

describe('handleDeviceTypeBasedIdentify function', () => {
  let writeSpy, navigation, readSpy;
  beforeEach(() => {
    writeSpy = jest.spyOn(BleManager, 'writeCharacteristicWithResponseForDevice')
    readSpy = jest.spyOn(BleManager, 'readCharacteristicForDevice')
    navigation = {navigate: jest.fn()};
    jest.spyOn(BleManager,'connectToDevice').mockResolvedValueOnce({discoverAllServicesAndCharacteristics: jest.fn(), })
  });
  it('should trigger identify device alarm on BLE device, when device type is in name itself', async () => {
    jest.spyOn(base64,'encode').mockImplementation(()=>{
      return 'beep-tone'
    }) 
    await store.dispatch(handleDeviceTypeBasedIdentify({id: 1, localName: 'Brx-asdkasbj', deviceType:DeviceType.ms700, nameWithType:'Brx-asdkasbj(MS700)' },navigation));
    expect(writeSpy).toBeCalledWith(1,UUIDMappingMS700.rootServiceUDID, UUIDMappingMS700.identifyDevice,'beep-tone');
  });
  it('should read device type from BLE and then perform Identify, when device type is not in the name, and the BLE is MS700', async () => {

    jest.spyOn(BleManager,'connectToDevice').mockResolvedValueOnce({discoverAllServicesAndCharacteristics: jest.fn()})
    jest.spyOn(BleManager,'readCharacteristicForDevice').mockResolvedValueOnce({value: 'TVMtNzAw'})
    jest.spyOn(base64,'decode').mockImplementation(()=>{
      return 'MS-700'
    })
    await store.dispatch(handleDeviceTypeBasedIdentify({id: 1, localName: 'Brx-asdkasbj', deviceType:DeviceType.infoView, nameWithType:'Brx-asdkasbj'},navigation,()=>{}));
    expect(readSpy).toBeCalledWith(1,UUIDMappingMS700.rootServiceUDID, UUIDMappingMS700.deviceType);
    expect(writeSpy).toBeCalledWith(1,UUIDMappingMS700.rootServiceUDID, UUIDMappingMS700.identifyDevice,'beep-tone');
  });
  it('should read device type from BLE and then perform Identify, when device type is not in the name, and the BLE is Infoview', async () => {
    jest.spyOn(base64,'encode').mockImplementation(()=>{
      return 'beep-tone'
    }) 
    jest.spyOn(base64,'encode').mockImplementation(()=>{
      return 'InfoviewPlayer'
    }) 
    await store.dispatch(handleDeviceTypeBasedIdentify({id: 1, localName: 'Brx-asdkasbj', deviceType:DeviceType.infoView, nameWithType:'Brx-asdkasbj' },navigation));
    expect(readSpy).toBeCalledWith(1,UUIDMappingMS700.rootServiceUDID, UUIDMappingMS700.deviceType);
    expect(writeSpy).toBeCalledWith(1,UUIDMappingMS700.rootServiceUDID, UUIDMappingMS700.identifyDevice,'beep-tone');
  });
  it('should read device type from BLE and then perform Identify, when device type is not in the name, and the BLE is CZA1300', async () => {
    jest.spyOn(base64,'encode').mockImplementation(()=>{
      return 'blink-leds'
    }) 
    jest.spyOn(base64,'encode').mockImplementation(()=>{
      return 'cza1300'
    }) 
    await store.dispatch(handleDeviceTypeBasedIdentify({id: 1, localName: 'Brx-asdkasbj', deviceType:DeviceType.cza1300, nameWithType:'Brx-asdkasbj' },navigation));
    expect(readSpy).toBeCalledWith(1,UUIDMappingMS700.rootServiceUDID, UUIDMappingMS700.deviceType);
    expect(writeSpy).toBeCalledWith(1,UUIDMappingMS700.rootServiceUDID, UUIDMappingMS700.identifyDevice,'blink-leds');
  });
});

describe('navigateToDevice function', () => {
  let writeSpy, navigation, readSpy, navigationSpy;
  beforeEach(() => {
    writeSpy = jest.spyOn(BleManager, 'writeCharacteristicWithResponseForDevice')
    readSpy = jest.spyOn(BleManager, 'readCharacteristicForDevice')
    store.dispatch(
      updateBleDevicesFields('bluetoothDevices', [{localName: 'Brx-emulator', deviceType:'',id:1}]),
    );
    navigation = {navigate: jest.fn()};
    navigationSpy = jest.spyOn(navigation,'navigate')
  });
  it('should navigate to BLE device setings screen', async () => {
    jest.spyOn(base64,'encode').mockImplementation(()=>{
      return 'beep-tone'
    }) 
    await store.dispatch(navigateToDevice({id: 1, localName: 'Brx-asdkasbj', deviceType:DeviceType.ms700, nameWithType:'Brx-asdkasbj(MS700)' },navigation));
    expect(navigationSpy).toBeCalledWith('HomeScreen');
  });
  it('should navigate to device setings screen, when CZA1300 device is pressed', async () => {
    jest.spyOn(base64,'encode').mockImplementation(()=>{
      return 'beep-tone'
    }) 
    await store.dispatch(navigateToDevice({id: 1, localName: 'Brx-asdkasbj', deviceType:DeviceType.cza1300, nameWithType:'Brx-asdkasbj(CZA-1300)' },navigation));
    expect(navigationSpy).toBeCalledWith('HomeScreen');
  });
  it('should read device type from BLE, then navigate to device settings screen', async () => {
    jest.spyOn(BleManager,'connectToDevice').mockResolvedValueOnce({discoverAllServicesAndCharacteristics: jest.fn()})
    jest.spyOn(BleManager,'readCharacteristicForDevice').mockResolvedValueOnce({value: 'TVMtNzAw'})
    jest.spyOn(base64,'decode').mockImplementation(()=>{
      return 'MS-700'
    })
    await store.dispatch(navigateToDevice({id: 1, localName: 'Brx-asdkasbj', deviceType:DeviceType.infoView, nameWithType:'Brx-asdkasbj' },navigation));
    expect(readSpy).toBeCalledWith(1,UUIDMappingMS700.rootServiceUDID, UUIDMappingMS700.deviceType);
    expect(navigationSpy).toBeCalledWith('HomeScreen');
  });
});


describe('check device type of devices found during the device discovery phase', () => {
  beforeEach(() => {
    const device = {
      id: 'BDCB6EA5-D458-DD27-E14E-169930912022',
      localName: 'Brx-emulator (CZA-1300)',
      name: 'osboxes',
      isConnected: false,
    };
    const navigation = {replace: () => jest.fn()};
    const callback = jest.fn();
    store.dispatch(handleDevicesFound(device,callback));
  });
  it('should check if the device found is CZA1300 device', () => {
    expect(store.getState().ble.bluetoothDevices[0].deviceType).toBe(deviceType.cza1300);
  });
});