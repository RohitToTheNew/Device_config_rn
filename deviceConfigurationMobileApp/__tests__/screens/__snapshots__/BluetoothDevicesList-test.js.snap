// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`on landing to bluetooth devices list should render devices list correctly 1`] = `null`;
